tests/reconfig/setup.md

```diff
@@ -0,0 +1,96 @@
+# Reconfig tests
+
```

Comment:
I'd recommend adding "Goals" section to explain the (high-level) goals of the tests -- that will help the reader get the idea why the test needed in the first place before they read the details.